### PR TITLE
Show Chef::VERSION at prompt_c and prompt_i on shell session

### PR DESCRIPTION
**Pull request by sawanoboly**
Hi, I want to always show version on shell session.

```
$ chef-shell 
loading configuration: none (standalone session)
Session type: standalone
Loading........done.

This is the chef-shell.
 Chef Version: 12.3.0.dev.0
 http://www.chef.io/
 http://docs.chef.io/

run `help' for help, `exit' or ^D to quit.

Ohai2u sawanoboriyu@!
chef (12.3.0.dev.0)> 
chef (12.3.0.dev.0)>   
chef (12.3.0.dev.0)>   recipe_mode
chef:recipe (12.3.0.dev.0)>   
```

### CODE REVIEW

File: lib/chef/shell.rb [hunk 1]
```diff
@@ -108,9 +108,9 @@ def self.configure_irb
     irb_conf[:IRB_RC] = lambda do |conf|
       m = conf.main
 
-      conf.prompt_c       = "chef#{leader(m)} > "
+      conf.prompt_c       = "chef#{leader(m)} (#{Chef::VERSION})> "
```
Comment (btm) [inline lib/chef/shell.rb]: Looking at http://ruby-doc.org/stdlib-2.0/libdoc/irb/rdoc/IRB.html, it seems like prompt_c "prompt for continuated statement" would be excessive?

**Comment by btm** [comment]
not sure about prompt_c, but :+1: otherwise.

**Comment by sawanoboly** [comment]
I've found one more idea of implement. 
We'll have `leader(m)` append "(#{Chef::VERSION})" by defaut.

That implement appends Chef::VERSION to all prompt, but users can overwrite by setting Shell.env .

For example, below.

```
  def self.configure_irb
    irb_conf[:HISTORY_FILE] = Chef::Util::PathHelper.home(".chef", "chef_shell_history")
    irb_conf[:SAVE_HISTORY] = 1000

    irb_conf[:IRB_RC] = lambda do |conf|
      m = conf.main

      conf.prompt_c       = "chef#{leader(m)} > "
      conf.return_format  = " => %s \n"
      conf.prompt_i       = "chef#{leader(m)} > "
      conf.prompt_n       = "chef#{leader(m)} ?> "
      conf.prompt_s       = "chef#{leader(m)}%l> "
      conf.use_tracer     = false
    end
  end

  def self.leader(main_object)
-    env_string = Shell.env ? " (#{Shell.env})" : ""
+    env_string = Shell.env ? " (#{Shell.env})" : "(#{Chef::VERSION})"
    LEADERS[main_object.class] + env_string
  end
```

Which idea do you like ?

- Just remove from prompt_c.
- make leader(m) returns by default.

**Comment by sawanoboly** [comment]
Ah, `Shell.env` includes `chef_environment`... ? I shoud not to use it.

**Comment by sawanoboly** [comment]
I've removed from prompt_c, Thanks :smile:

**Comment by btm** [comment]
@chef/client-core review?

**Comment by jaymzh** [comment]
Love it. :+1:

**Comment by danielsdeleo** [comment]
:+1:  on the code change. Personally, I'm ambivalent on the change so if other people think it's cool then let's ship it.

**Comment by tyler-ball** [comment]
:+1: Lets do it unless someone doesn't like it

**Comment by ranjib** [comment]
:+1:

**Pull request merged**